src/android/registerNativeModule.js

```diff
@@ -9,9 +11,15 @@ const applyPatch = (filePath, patch) =>
 
 module.exports = function registerNativeAndroidModule(name, dependencyConfig, projectConfig) {
   const prefix = getPrefix(getReactVersion(projectConfig.folder));
-  const makeSettingsPatch = require(`./patches/makeSettingsPatch`);
-  const makeBuildPatch = require(`./patches/makeBuildPatch`);
-  const makeMainActivityPatch = require(`./${prefix}/makeMainActivityPatch`);
+
+  try {
+    const makeSettingsPatch = require(`./patches/makeSettingsPatch`);
+    const makeBuildPatch = require(`./patches/makeBuildPatch`);
+    const makeMainActivityPatch = require(`./${prefix}/makeMainActivityPatch`);
+  } catch (e) {
+    log.error(e + '.\nIt seems somethins went wrong while patching ' +
+      'Android modules.\nPlease file an issue here: ' + pkg.bugs.url);
+  }
```

Comment:
Unfortunately, the reason why I made it - is because you can't propagate anything from the dynamically required module up. Otherwise it wouldn't be even needed.